src/pages/index.js

```diff
@@ -40,89 +50,222 @@ const enableValidation = (config) => {
+    });
+  avatarEditPopup.close();
+}
+
+avatarEditButton.addEventListener("click", () => {
+  avatarEditPopup.open();
+});
+
+/* -------------------------------------------------------------------------- */
+/*                                    Card                                    */
+/* -------------------------------------------------------------------------- */
+/* ---------------------------------- Card ---------------------------------- */
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   const cardElement = card.generateCard();
   return cardElement;
 }
+/* -------------------------------- CardList -------------------------------- */
+const cardList = new Section(
+  {
+    renderer: (items) => {
+      cardList.addItem(items);
+    },
+  },
+  ".cards__list"
+);
 
-function renderCard(cardData) {
-  const cardElement = createCard(cardData);
-  cardList.addItem(cardElement);
-}
-/* -------------------------------------------------------------------------- */
-/*                               Event Handlers                               */
-/* -------------------------------------------------------------------------- */
-/* ------------------------------ Profile Edit ------------------------------ */
-function handleProfileEditSubmit(userData) {
-  profileInfo.setUserInfo(userData);
-  profileEditPopup.close();
+function cardListData() {
+  return api
+    .getInitialCards()
+    .then((res) => {
+      return res.map((card) => ({
+        name: card.name,
+        link: card.link,
+        _id: card._id,
+      }));
+    })
+    .catch((err) => {
+      console.error(err);
+      return [];
+    });
 }
+
+cardListData()
+  .then((data) => {
+    console.log("data", data);
+    const cardElement = data.map((cardData) => createCard(cardData));
+    cardList.renderItems(cardElement);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+/* ------------------------------ cardAddPopup ------------------------------ */
+
+const cardAddPopup = new PopupWithForm({
+  popupSelector: "#card-add-modal",
+  handleFormSubmit: handleCardAddSubmit,
+});
+cardAddPopup.setEventListeners();
+
+cardAddButton.addEventListener("click", () => {
+  cardAddPopup.open();
+});
+
 function handleCardAddSubmit(formValues) {
   const name = formValues.title;
   const link = formValues.link;
-  renderCard({ name, link });
+  cardAddPopup.renderLoading(true);
+  api
+    .addNewCard({ name, link })
+
+    .then((card) => {
+      const cardElement = createCard(card);
+      cardList.addItem(cardElement);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      cardAddPopup.renderLoading(false);
+    });
+
   cardAddPopup.close();
   cardAddForm.reset();
   formValidators[cardAddForm.getAttribute("id")].disableButton();
```

Comment:
You need to **reset** the form only in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.


And close, and disable in `then` as well